manifests/timer.pp

```diff
@@ -72,6 +75,7 @@
   Boolean                                  $show_diff = true,
   Optional[Variant[Boolean, Enum['mask']]] $enable = undef,
   Optional[Boolean]                        $active = undef,
+  Boolean                                  $daemon_reload = true,
```

Comment:
If `$active == true`, the daemon-reload is redundant, right? Wouldn't Puppet itself handle this since it inspects whether a reload is needed before (re)starting a service? And `$active == true` implies the service is managed.

Reply:
The daemon-reload generally needs to be sandwiched between a file drop and a subsequent service.

There's no good way to ensure that we only have to call it once without a very high risk of dependency loops.

Reply:
For `active` vs `daemon_reload`:

* We need to call `daemon_reload` after the file no matter what
* If the service is managed, we need to call it before the service
* If, for whatever reason, you *don't* want to call it (I don't know why, but I was trying to preserve the current state if users wanted it) then we need to do neither
That's what's currently implemented in `unit_file.pp`

Reply:
@ekohl Puppet might be doing fine but items like GDM and NFS need refreshes at very specific times which may be prior to the service loading.

Reply:
At any rate, I couldn't get it to work consistently without this chain in place. Might be able to debug later but this fixes several items for now.

Reply:
Convo bounced through IRC and into https://github.com/voxpupuli/puppet-systemd/issues/284